pandas/core/algorithms.py

```diff
@@ -1678,3 +1679,73 @@ def union_with_duplicates(
         unique_vals = ensure_wrapped_if_datetimelike(unique_vals)
     repeats = final_count.reindex(unique_vals).values
     return np.repeat(unique_vals, repeats)
+
+
+def map_array(arr: ArrayLike, mapper, na_action=None):
```

Comment:
Could you type na_action here?

Reply:
I've pushed a new version with typed parameters.